metasettings/models.py

```diff
@@ -83,6 +85,47 @@ def get_currency_rates(self, year=None, month=None):
 
         return self.default_rates
 
+    def update_or_create(self, currency, rate, date=None):
+        """Update a CurrencyRate object referenced by `currency` (and optionally
+        `date`). If the object is not found, a new one will be created.
+
+        Returns a tuple with the object and a `created` flag. If the object
+        was neither updated nor created then None is returned.
+
+        """
+        if currency not in self.currency_choices:
+            return (None, False)
+
+        try:
+            filters = {'currency': currency}
+            if date:
+                filters.update({
+                    'year': date.year,
+                    'month': date.month
+                })
+
+            existing_rate = CurrencyRate.objects.get(**filters)
+
+            existing_rate.rate = str(existing_rate.rate)
+            rate = str("%.2f" % (rate))
+
+            if existing_rate.rate == rate:
+                return (None, False)
+            else:
+                existing_rate.rate = rate
+                existing_rate.save()
+                return (existing_rate, False)
```

Comment:
could be:

```python
if existing.rate != rate:
    existing_rate.rate = rate
    existing_rate.save()
return existing_rate, False
```

to avoid the ``else`` clause.